api/src/OpenTelemetry/Logging/Core.hs

```diff
@@ -15,19 +28,73 @@ import Control.Applicative
 import Control.Monad.Trans
 import Control.Monad.Trans.Maybe
 import Data.Coerce
+import Data.IORef
 import Data.Maybe
+import GHC.IO (unsafePerformIO)
 import OpenTelemetry.Common
 import OpenTelemetry.Context
 import OpenTelemetry.Context.ThreadLocal
+import OpenTelemetry.Internal.Common.Types
 import OpenTelemetry.Internal.Logging.Types
 import OpenTelemetry.Internal.Trace.Types
+import OpenTelemetry.Resource (MaterializedResources, emptyMaterializedResources)
 import System.Clock
 
 
 getCurrentTimestamp :: (MonadIO m) => m Timestamp
 getCurrentTimestamp = liftIO $ coerce @(IO TimeSpec) @(IO Timestamp) $ getTime Realtime
 
 
+data LoggerProviderOptions = LoggerProviderOptions {loggerProviderOptionsResource :: MaterializedResources}
+
+
+{- | Options for creating a @LoggerProvider@ with no resources and default limits.
+
+ In effect, logging is a no-op when using this configuration.
```

Comment:
Trying to orient myself to this: what is it about this that makes logging a no-op? Is it that this is part of the API and thus not tied to a concrete implementation?

Reply:
That is a great question. I'm a little foggy about this idea myself, but I think that it means that it has no `Processors` or `Exporters` or any way to interact with the outside world. The API is supposed to be completely [no-op](https://opentelemetry.io/docs/specs/otel/logs/noop/) when the SDK has not registered a global LoggerProvider (See the second paragraph of `api/Readme.md`).

This got me thinking, so I'm working on making a way to make it clearer when things are no-op.

Reply:
Okay. Logging is no-op when the `LoggerProvider` has no processors. I think I will make a `noOpLoggerProvider` value in a later pull request to make this more clear.